x/sqlite.ts

```diff
@@ -0,0 +1,105 @@
+import * as sqlite from "https://deno.land/x/sqlite@v2.3.2/mod.ts";
+
+import * as driver from "../sql/driver.ts";
+
+export type Value = null | boolean | number | string | bigint | Uint8Array;
+
+export type Meta = driver.Meta<{
+  Value: Value;
+}>;
+
+export class Driver
+  implements driver.Driver<Meta>, driver.ConnectorOpener<Meta> {
+  openConnectorSync(path: string) {
+    return new Connector(this, path);
+  }
+}
+
+export class Connector implements driver.Connector<Meta> {
+  constructor(
+    readonly driver: Driver,
+    private readonly path: string,
+  ) {}
+
+  connectSync() {
+    const handle = new sqlite.DB(this.path);
+    return new Connection(this.driver, handle);
+  }
+}
+
+export class Connection implements driver.Connection<Meta> {
+  private activeTransaction?: Transaction;
+
+  constructor(
+    readonly driver: Driver,
+    private readonly handle: sqlite.DB,
+  ) {}
+
+  startTransactionSync() {
+    const transaction = new Transaction(this.driver, this.handle, this);
+    this.handle.query("BEGIN DEFERRED TRANSACTION").return();
+    return transaction;
+  }
+
+  lastInsertedIdSync(): Value {
+    return this.handle.lastInsertRowId;
+  }
+
+  affectedRowsSync(): number {
+    return this.handle.changes;
+  }
+
+  closeSync() {
+    this.handle.close();
+  }
+}
+
+export class Transaction implements driver.Transaction<Meta> {
+  private childTransaction?: Transaction;
+
+  constructor(
+    readonly driver: Driver,
+    private readonly connectionHandle: sqlite.DB,
+    parent: Transaction | Connection,
+  ) {
+    if (parent instanceof Transaction) {
+      this.name = `${parent.name}_${Transaction.nextId}`;
+    } else {
+      this.name = `transaction_${Transaction.nextId}`;
+    }
+    Transaction.nextId += 1;
+  }
+
+  commitSync(): undefined {
+    this.connectionHandle.query(`RELEASE ${this.name}`).return();
+    return;
+  }
+
+  rollbackSync(): undefined {
+    this.connectionHandle.query(`ROLLBACK TO ${this.name}`).return();
+    return;
+  }
+}
+
+/** A prepared statement, bound in the context of a parent transaction or connection. */
+export class PreparedStatement implements driver.PreparedStatement {
```

Comment:
There is actually a way to properly prepare statements such that the statement is reusable with different input, but this might be out-of-scope for this first version 😅

Reply:
Ah, I didn't think that was supported by `deno-sqlite`. If it is, maybe we'll go ahead and use that instead of shimming it.

Reply:
Well, it's not exactly supported. The `sqlite` api itself works this way, but it's abstracted away in the `query`. But if you are happy to talk to the underlying `WASM` api, you can do this:
- https://github.com/dyedgreen/deno-sqlite/blob/master/src/db.ts#L118
- https://github.com/dyedgreen/deno-sqlite/blob/master/build/src/wrapper.c#L74

But maybe this should be implemented directly in the `deno-sqlite` repo itself 😅